packages/backend-common/src/reading/GithubUrlReader.ts

```diff
@@ -180,4 +133,59 @@ export class GithubUrlReader implements UrlReader {
     const { host, token } = this.config;
     return `github{host=${host},authed=${Boolean(token)}}`;
   }
+
+  private async getRepoDetails(
+    url: string,
+  ): Promise<{
+    repo: RestEndpointMethodTypes['repos']['get']['response']['data'];
+    branch: RestEndpointMethodTypes['repos']['getBranch']['response']['data'];
```

Comment:
yeah it still does fetch instead of constructing an octokit, we may do that switch some day too if we get around to it